src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,14 +1,35 @@
 import React from 'react';
+import { actions as filterActions } from '../../features/filter';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { Status } from '../../types/Status';
 
 export const TodoFilter: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const query = useAppSelector(state => state.filter.query);
+
+  const handleStatusChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    dispatch(filterActions.setStatus(event.target.value as Status));
+  };
+
+  const handleQueryInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    dispatch(filterActions.setQuery(event.target.value));
+  };
+
+  const handleQueryRemoval = () => {
+    dispatch(filterActions.setQuery(''));
+  };
+
   return (
     <form
       className="field has-addons"
       onSubmit={event => event.preventDefault()}
     >
       <p className="control">
         <span className="select">
-          <select data-cy="statusSelect">
+          <select
+            data-cy="statusSelect"
+            onChange={handleStatusChange}
+          >
             <option value="all">All</option>
             <option value="active">Active</option>
             <option value="completed">Completed</option>
```

Comment:
You can render this using your Status enum `{
Object.entries(Status).map(([key, value]) => (
              <option
                value={value}
                key={key}
              >
                {key}
              </option>
            ))}
`